data/actions/roshamuul/strange_vortex_tp.lua

```diff
@@ -0,0 +1,19 @@
+local nightmarePosition = Position(33542, 32421, 15)
+
+local nightmareVortex = MoveEvent()
+
+function nightmareVortex.onStepIn(creature, item, position, fromPosition)
+    local player = creature:getPlayer()
+    if not player then
+        return false
+    end
+
+    player:teleportTo(nightmarePosition)
+    player:sendTextMessage(MESSAGE_EVENT_ADVANCE, "You have been trapped in a nightmare.")
+
+    return true
+end
+
+nightmareVortex:type("stepin")
+nightmareVortex:aid(33542)
```

Comment:
```suggestion
nightmareVortex:position({x = posx, y = posy, z = posz})
```
Change the position to the position of the tile where the action needs to be placed

Reply:
The tile where actions need to be placed is where the Minion of Gaz'haragoth dies. I couldn't do this only in Creaturescripts because item 20121 is not a teleport.